memory-viz/src/memory_model.ts

```diff
@@ -59,15 +61,22 @@ export class MemoryModel {
     prop_min_height: number = 50; // Minimum height of type and id boxes
     obj_x_padding: number = 25; // Minimum horizontal padding of object rectangle
     canvas_padding: number = 25; // Minimum padding of the canvas
+    canvas_bottom_padding: number = 100; // Minimum padding of the bottom of the canvas
     double_rect_sep: number = 6; // Separation between double boxes around immutable objects
     list_index_sep: number = 20; // Vertical offset for list index labels
     font_size: number = 20; // Font size, in px
     browser: boolean = false; // Whether this library is being used in a browser context
     roughjs_config: Config; // Configuration object used to pass in options to rough.js
     width?: number; // Width of the canvas, dynamically updated if not provided in options
     height?: number; // Height of the canvas, dynamically updated if not provided in options
+    manual_height: boolean = false;
```

Comment:
This shouldn't be an additional parameter (it should instead just be derived from existing parameters).